packages/react/src/utils/useAnchorPositioning.ts

```diff
@@ -118,12 +138,31 @@ export function useAnchorPositioning(
   // presence.
   const arrowRef = React.useRef<Element | null>(null);
 
+  // Keep these reactive if they're not functions
+  const sideOffsetDep = typeof sideOffset !== 'function' ? sideOffset : 0;
+  const alignOffsetDep = typeof alignOffset !== 'function' ? alignOffset : 0;
+
   const middleware: UseFloatingOptions['middleware'] = [
-    offset({
-      mainAxis: sideOffset,
-      crossAxis: alignOffset,
-      alignmentAxis: alignOffset,
-    }),
+    offset(
+      ({ rects, placement: currentPlacement }) => {
+        const data = {
+          side: getLogicalSide(sideParam, getSide(currentPlacement), isRtl),
+          align: getAlignment(currentPlacement) || 'center',
+          anchor: { width: rects.reference.width, height: rects.reference.height },
+          popup: { width: rects.floating.width, height: rects.floating.height },
```

Comment:
Technically this is the `Positioner` element, not `Popup`, but in practically every case they should be the same since it's a presentational wrapper